### PR TITLE
feat: update deps and switch to klog

### LINKED ISSUE #347: Switch to klog
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
Switch to klog for logging so we can get features like json logging.

- [ ] Update to using klog.
- [ ] Audit current logging to ensure there are sufficient logs indicating current status/errors.
- [ ] Configurable log verbosity

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Kubernetes version: (use `kubectl version`):

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:
- Switches to using `klog` for logging instead of `logrus`
  - Adapts structured logging - https://kubernetes.io/blog/2020/09/04/kubernetes-1-19-introducing-structured-logs/
- Deprecates debug flag as klog log level is set using `-v`
- Adds option to configure json logging format
- Updates client-go deps to `0.19.2` and controller-runtime to `0.6.3`
- Regenrates clientset, informers and listers

/kind feature

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #347 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by aramase** [comment]
@ritazh @tam7t @tomhjp PTAL at this PR when you get a chance. Switches to using `klog` with structured logging.

**Review by tam7t — APPROVED** [review]
LGTM, I like the changes!

File: controllers/secretproviderclasspodstatus_controller.go [hunk 1]
```diff
@@ -184,76 +181,76 @@ func (r *SecretProviderClassPodStatusReconciler) Reconcile(req ctrl.Request) (ct
...
 		if err = secretutil.ValidateSecretObject(*secretObj); err != nil {
-			logger.Errorf("failed to validate secret object in spc %s/%s, err: %+v", spc.Namespace, spc.Name, err)
+			klog.ErrorS(err, "failed to validate secret object in spc", "spc", klog.KObj(spc), "spcps", klog.KObj(spcPodStatus))
 			errs = append(errs, fmt.Errorf("failed to validate secret object in spc %s/%s, err: %+v", spc.Namespace, spc.Name, err))
 			continue
 		}
 		exists, err := r.secretExists(ctx, secretName, req.Namespace)
 		if err != nil {
-			logger.Errorf("failed to check if secret %s exists, err: %+v", secretName, err)
+			klog.ErrorS(err, "failed to check if secret exists", "secret", secretName, "spcps", klog.KObj(spcPodStatus))
```
[the remark this answers is not in the document]
Comment (aramase) [reply controllers/secretproviderclasspodstatus_controller.go]: Done!

File: controllers/secretproviderclasspodstatus_controller.go [hunk 2]
```diff
@@ -282,7 +279,7 @@ func (r *SecretProviderClassPodStatusReconciler) Reconcile(req ctrl.Request) (ct
 
 			createFn := func() (bool, error) {
 				if err := r.createK8sSecret(ctx, secretName, req.Namespace, datamap, labelsMap, secretType); err != nil {
-					logger.Errorf("failed createK8sSecret, err: %v for secret: %s", err, secretName)
+					klog.ErrorS(err, "failed to create Kubernetes secret", "secret", secretName, "spcps", klog.KObj(spcPodStatus))
```
[the remark this answers is not in the document]
Comment (aramase) [reply controllers/secretproviderclasspodstatus_controller.go]: Done!

File: cmd/secrets-store-csi-driver/main.go [hunk 3]
```diff
@@ -41,9 +42,7 @@ var (
 	endpoint           = flag.String("endpoint", "unix://tmp/csi.sock", "CSI endpoint")
 	driverName         = flag.String("drivername", "secrets-store.csi.k8s.io", "name of the driver")
 	nodeID             = flag.String("nodeid", "", "node id")
-	debug              = flag.Bool("debug", false, "sets log to debug level")
```
Comment (ritazh) [inline cmd/secrets-store-csi-driver/main.go:44]: Do we want to introduce a breaking change here or should we return a warning that this as deprecated is provided?

File: pkg/secrets-store/nodeserver.go [hunk 4]
```diff
@@ -105,33 +106,32 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
...
+	podUID = attrib[csipoduid]
+
 	mounted, err = ns.ensureMountPoint(targetPath)
 	if err != nil {
 		errorReason = internalerrors.FailedToEnsureMountPoint
 		return nil, status.Errorf(codes.Internal, "Could not mount target %q: %v", targetPath, err)
 	}
 	if mounted {
-		log.Infof("NodePublishVolume: %s is already mounted", targetPath)
+		klog.InfoS("target path is already mounted", "targetPath", targetPath, "pod", podName)
```
Comment (ritazh) [inline pkg/secrets-store/nodeserver.go:121]: do we need the namespace here too?

File: pkg/secrets-store/nodeserver.go [hunk 5]
```diff
@@ -105,33 +106,32 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
...
-	podNamespace = attrib[csipodnamespace]
-	podUID = attrib[csipoduid]
+	klog.V(2).InfoS("node publish volume", "target", targetPath, "volumeId", volumeID, "attributes", attrib, "mountflags", mountFlags)
 
 	if isMockProvider(providerName) {
 		// mock provider is used only for running sanity tests against the driver
 		err := ns.mounter.Mount("tmpfs", targetPath, "tmpfs", []string{})
 		if err != nil {
-			log.Errorf("mount err: %v for pod: %s, ns: %s", err, podUID, podNamespace)
+			klog.ErrorS(err, "failed to mount", "pod", podName)
```
Comment (ritazh) [inline pkg/secrets-store/nodeserver.go:131]: do we need the namespace too?

File: cmd/secrets-store-csi-driver/main.go (hunk 3 above)
Comment (aramase) [reply cmd/secrets-store-csi-driver/main.go:44]: Done!

File: pkg/secrets-store/nodeserver.go (hunk 4 above)
Comment (aramase) [reply pkg/secrets-store/nodeserver.go]: Updated!

File: pkg/secrets-store/nodeserver.go (hunk 5 above)
Comment (aramase) [reply pkg/secrets-store/nodeserver.go]: Updated!

**Comment by aramase** [comment]
```
W1103 19:54:33.628501       1 main.go:75] --debug flag has been DEPRECATED and will be removed in future releases. Use -v=<log level> to configure log verbosity.
I1103 19:54:33.628557       1 exporter.go:35] metrics backend: prometheus
I1103 19:54:33.932774       1 main.go:111] starting manager
I1103 19:54:34.125354       1 reconciler.go:122] starting rotation reconciler with poll interval: 10s
I1103 19:54:34.429055       1 secrets-store.go:97] Driver: secrets-store.csi.k8s.io
I1103 19:54:34.429084       1 secrets-store.go:98] Version: klog1103
I1103 19:54:34.429090       1 secrets-store.go:99] Provider Volume Path: /etc/kubernetes/secrets-store-csi-providers
I1103 19:54:34.429095       1 secrets-store.go:100] Minimum provider versions:
I1103 19:54:34.429101       1 secrets-store.go:101] GRPC supported providers: azure
I1103 19:54:34.429125       1 driver.go:82] "Enabling controller service capability" capability="CREATE_DELETE_VOLUME"
I1103 19:54:34.429149       1 driver.go:92] "Enabling volume access mode" mode="SINGLE_NODE_READER_ONLY"
I1103 19:54:34.429157       1 driver.go:92] "Enabling volume access mode" mode="MULTI_NODE_READER_ONLY"
I1103 19:54:34.429326       1 secrets-store.go:65] minimum compatible provider versions not specified with --min-provider-version
I1103 19:54:34.429531       1 server.go:107] Listening for connections on address: //csi/csi.sock
I1103 19:55:42.519632       1 nodeserver.go:303] "Using grpc client" provider="azure" pod="demo-pod"
I1103 19:55:45.172183       1 nodeserver.go:204] "node publish volume complete" targetPath="/var/lib/kubelet/pods/9ed22c28-c855-446a-abd4-27abedb2dc86/volumes/kubernetes.io~csi/secrets-store-inline/mount" pod="default/demo-pod"
I1103 19:55:45.172279       1 secretproviderclasspodstatus_controller.go:184] "reconcile started" spcps="default/demo-pod-default-azure-kv"
I1103 19:55:45.272632       1 secretproviderclasspodstatus_controller.go:223] "no secret objects defined for spc, nothing to reconcile" spc="default/azure-kv" spcps="default/demo-pod-default-azure-kv"
I1103 19:55:55.023670       1 reconciler.go:334] "spc doesn't contain secret objects" spc="default/azure-kv" pod="default/demo-pod" controller="rotation"
I1103 19:56:04.624088       1 reconciler.go:334] "spc doesn't contain secret objects" spc="default/azure-kv" pod="default/demo-pod" controller="rotation"
I1103 19:56:14.412594       1 reconciler.go:334] "spc doesn't contain secret objects" spc="default/azure-kv" pod="default/demo-pod" controller="rotation"
I1103 19:56:24.524138       1 reconciler.go:334] "spc doesn't contain secret objects" spc="default/azure-kv" pod="default/demo-pod" controller="rotation"
I1103 19:56:30.891735       1 secretproviderclasspodstatus_controller.go:184] "reconcile started" spcps="default/demo-pod-default-azure-kv"
I1103 19:56:30.891786       1 secretproviderclasspodstatus_controller.go:189] "reconcile complete" spcps="default/demo-pod-default-azure-kv"
I1103 19:56:32.698088       1 nodeserver.go:254] "node unpublish volume complete" targetPath="/var/lib/kubelet/pods/9ed22c28-c855-446a-abd4-27abedb2dc86/volumes/kubernetes.io~csi/secrets-store-inline/mount" podUID="9ed22c28-c855-446a-abd4-27abedb2dc86"
I1103 19:57:07.128802       1 nodeserver.go:303] "Using grpc client" provider="azure" pod="nginx-68c67c5b5c-2x6dz"
I1103 19:57:07.828733       1 nodeserver.go:204] "node publish volume complete" targetPath="/var/lib/kubelet/pods/e4385fe3-b8a3-4a64-b22e-5a96d2eaed39/volumes/kubernetes.io~csi/secrets-store-inline/mount" pod="default/nginx-68c67c5b5c-2x6dz"
I1103 19:57:07.828759       1 secretproviderclasspodstatus_controller.go:184] "reconcile started" spcps="default/nginx-68c67c5b5c-2x6dz-default-azure"
I1103 19:57:08.031832       1 secretproviderclasspodstatus_controller.go:351] "successfully created Kubernetes secret" secret="default/foosecret"
I1103 19:57:08.031867       1 secretproviderclasspodstatus_controller.go:323] "reconcile complete" spc="default/azure" pod="default/nginx-68c67c5b5c-2x6dz" spcps="default/nginx-68c67c5b5c-2x6dz-default-azure"
I1103 19:57:54.621014       1 reconciler.go:305] "updating versions in spc pod status" spcps="default/nginx-68c67c5b5c-2x6dz-default-azure" controller="rotation"
I1103 19:57:54.627755       1 secretproviderclasspodstatus_controller.go:184] "reconcile started" spcps="default/nginx-68c67c5b5c-2x6dz-default-azure"
I1103 19:57:54.627985       1 secretproviderclasspodstatus_controller.go:323] "reconcile complete" spc="default/azure" pod="default/nginx-68c67c5b5c-2x6dz" spcps="default/nginx-68c67c5b5c-2x6dz-default-azure"
I1103 19:58:50.798594       1 nodeserver.go:254] "node unpublish volume complete" targetPath="/var/lib/kubelet/pods/e4385fe3-b8a3-4a64-b22e-5a96d2eaed39/volumes/kubernetes.io~csi/secrets-store-inline/mount" podUID="e4385fe3-b8a3-4a64-b22e-5a96d2eaed39"
E1103 19:58:53.933237       1 secretproviderclasspodstatus_controller.go:98] "failed to patch secret owner ref" err="failed to get spc azure, err: SecretProviderClass.secrets-store.csi.x-k8s.io \"azure\" not found"
E1103 19:58:54.125617       1 reconciler.go:492] "failed to reconcile spc for pod" err="failed to get secret provider class default/azure, err: secretproviderclasses.secrets-store.csi.x-k8s.io \"default/azure\" not found" spc="azure" pod="nginx-68c67c5b5c-2x6dz" controller="rotation"
E1103 19:58:58.933080       1 secretproviderclasspodstatus_controller.go:98] "failed to patch secret owner ref" err="failed to get spc azure, err: SecretProviderClass.secrets-store.csi.x-k8s.io \"azure\" not found"
I1103 19:59:00.392380       1 secretproviderclasspodstatus_controller.go:184] "reconcile started" spcps="default/nginx-68c67c5b5c-2x6dz-default-azure"
I1103 19:59:00.392437       1 secretproviderclasspodstatus_controller.go:189] "reconcile complete" spcps="default/nginx-68c67c5b5c-2x6dz-default-azure"
I1103 19:59:04.282272       1 reconciler.go:519] "retry budget exceeded, dropping from queue" spcps="default/nginx-68c67c5b5c-2x6dz-default-azure"
```

**Pull request merged**